File: app/src/main/java/com/curiousily/ipoli/app/receivers/BootCompleteReceiver.java
package com.curiousily.ipoli.app.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.curiousily.ipoli.Constants;
import com.curiousily.ipoli.app.jobs.RemindPlanDayJob;
import com.curiousily.ipoli.utils.Time;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 9/21/15.
 */
public class BootCompleteReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {

        Time time = Time.of(Constants.DEFAULT_PLAN_DAY_TIME);
        new RemindPlanDayJob(context, time).schedule();
    }
}

File: app/src/main/java/io/ipoli/android/common/BootCompleteReceiver.kt
package io.ipoli.android.common

import android.content.Context
import android.content.Intent
import io.ipoli.android.common.notification.QuickDoNotificationUtil
import io.ipoli.android.common.view.AppWidgetUtil
import kotlinx.coroutines.experimental.android.UI
import kotlinx.coroutines.experimental.launch
import org.threeten.bp.LocalDate
import space.traversal.kapsule.required

/**
 * Created by Venelin Valkov <venelin@mypoli.fun>
 * on 03/03/2018.
 */
class BootCompleteReceiver : AsyncBroadcastReceiver() {

    private val playerRepository by required { playerRepository }
    private val questRepository by required { questRepository }
    private val reminderScheduler by required { reminderScheduler }

    override suspend fun onReceiveAsync(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val p = playerRepository.find()
            if (p != null) {
                reminderScheduler.schedule()
                launch(UI) {
                    AppWidgetUtil.updateAgendaWidget(context)
                    AppWidgetUtil.updateHabitWidget(context)
                    if (p.isDead) {
                        QuickDoNotificationUtil.showDefeated(context)
                    } else {
                        val quests = questRepository.findScheduledAt(LocalDate.now())
                        if (p.preferences.isQuickDoNotificationEnabled) {
                            QuickDoNotificationUtil.update(
                                context = context,
                                quests = quests
                            )
                        }
                    }
                }
            }
        }
    }

}